src/coreComponents/constitutive/fluid/multifluid/compositional/functions/NegativeTwoPhaseFlash_impl.hpp

```diff
@@ -0,0 +1,182 @@
+
+  KValueInitialization::computeWilsonGasLiquidKvalue( numComps,
+                                                      pressure,
+                                                      temperature,
+                                                      criticalPressure,
+                                                      criticalTemperature,
+                                                      acentricFactor,
+                                                      kVapourLiquid );
+
+  bool converged = false;
+  for( localIndex iterationCount = 0; iterationCount < maxIterations; ++iterationCount )
+  {
+    // Solve Rachford-Rice Equation
+    vapourPhaseMoleFraction = RachfordRice::solve( kVapourLiquid, composition, presentComponentIds );
+
+    // Assign phase compositions
+    for( integer const ic : presentComponentIds )
+    {
+      liquidComposition[ic] = composition[ic] / ( 1.0 + vapourPhaseMoleFraction * ( kVapourLiquid[ic] - 1.0 ) );
+      vapourComposition[ic] = kVapourLiquid[ic] * liquidComposition[ic];
+    }
+
+    normalizeComposition( numComps, liquidComposition );
+    normalizeComposition( numComps, vapourComposition );
+
+    // Compute the phase fugacities
+    CubicEOSPhaseModel< EOS_TYPE_LIQUID >::compute( numComps,
+                                                    pressure,
+                                                    temperature,
+                                                    liquidComposition,
+                                                    criticalPressure,
+                                                    criticalTemperature,
+                                                    acentricFactor,
+                                                    binaryInteractionCoefficients,
+                                                    logLiquidFugacity );
+    CubicEOSPhaseModel< EOS_TYPE_VAPOUR >::compute( numComps,
+                                                    pressure,
+                                                    temperature,
+                                                    vapourComposition,
+                                                    criticalPressure,
+                                                    criticalTemperature,
+                                                    acentricFactor,
+                                                    binaryInteractionCoefficients,
+                                                    logVapourFugacity );
+
+    // Compute fugacity ratios and check convergence
+    converged = true;
+
+    for( integer const ic : presentComponentIds )
+    {
+      fugacityRatios[ic] = exp( logLiquidFugacity[ic] - logVapourFugacity[ic] ) * liquidComposition[ic] / vapourComposition[ic];
+      if( fugacityTolerance < fabs( fugacityRatios[ic] - 1.0 ) )
+      {
+        converged = false;
+      }
+    }
+
+    if( converged )
+    {
+      break;
+    }
+
+    // Update K-values
+    for( auto ic : presentComponentIds )
```

Comment:
```suggestion
    for( integer const ic : presentComponentIds )
```
to conform to the style currently used in the code